frontend/model/contracts/group.js

```diff
@@ -456,29 +458,36 @@ sbp('chelonia/defineContract', {
   actions: {
     // this is the constructor
     'gi.contracts/group': {
-      validate: objectMaybeOf({
-        settings: objectMaybeOf({
-          // TODO: add 'groupPubkey'
-          groupName: string,
-          groupPicture: unionOf(string, objectOf({
-            manifestCid: string,
-            downloadParams: optional(object)
-          })),
-          sharedValues: string,
-          mincomeAmount: number,
-          mincomeCurrency: string,
-          distributionDate: isPeriodStamp,
-          distributionPeriodLength: number,
-          minimizeDistribution: boolean,
-          proposals: objectOf({
-            [PROPOSAL_INVITE_MEMBER]: proposalSettingsType,
-            [PROPOSAL_REMOVE_MEMBER]: proposalSettingsType,
-            [PROPOSAL_GROUP_SETTING_CHANGE]: proposalSettingsType,
-            [PROPOSAL_PROPOSAL_SETTING_CHANGE]: proposalSettingsType,
-            [PROPOSAL_GENERIC]: proposalSettingsType
+      validate: (data) => {
+        objectMaybeOf({
+          settings: objectMaybeOf({
+            // TODO: add 'groupPubkey'
+            groupName: string,
+            groupPicture: unionOf(string, objectOf({
+              manifestCid: string,
+              downloadParams: optional(object)
+            })),
+            sharedValues: string,
+            mincomeAmount: number,
+            mincomeCurrency: string,
+            distributionDate: isPeriodStamp,
+            distributionPeriodLength: number,
+            minimizeDistribution: boolean,
+            proposals: objectOf({
+              [PROPOSAL_INVITE_MEMBER]: proposalSettingsType,
+              [PROPOSAL_REMOVE_MEMBER]: proposalSettingsType,
+              [PROPOSAL_GROUP_SETTING_CHANGE]: proposalSettingsType,
+              [PROPOSAL_PROPOSAL_SETTING_CHANGE]: proposalSettingsType,
+              [PROPOSAL_GENERIC]: proposalSettingsType
+            })
           })
-        })
-      }),
+        })(data)
+
+        const groupName = data.settings.groupName
+        if (!stringMax(GROUP_NAME_MAX_CHAR)(groupName)) {
+          throw new TypeError(`A group name cannot exceed ${GROUP_NAME_MAX_CHAR} characters.`)
+        }
+      },
```

Comment:
This could be restored to how it was and made way more secure if you implement `stringMax()` validation function (#902).

Then there's no need to create a custom validation function, and every one of `string` parameters here could be validated.